Microsoft.Azure.Cosmos/src/Handler/TelemetryHandler.cs

```diff
@@ -40,11 +40,12 @@ public TelemetryHandler(ClientTelemetry telemetry)
                                 resourceType: request.ResourceType,
                                 consistencyLevel: request.Headers?[Documents.HttpConstants.HttpHeaders.ConsistencyLevel],
                                 requestCharge: response.Headers.RequestCharge,
-                                subStatusCode: response.Headers.SubStatusCode);
+                                subStatusCode: response.Headers.SubStatusCode,
+                                trace: response.Trace);
                 }
                 catch (Exception ex)
                 {
-                    DefaultTrace.TraceError("Error while collecting telemetry information : " + ex.Message);
+                    DefaultTrace.TraceError($"Error while collecting telemetry information : {ex}");
```

Comment:
Don't materialize, use arguments.

```suggestion
                    DefaultTrace.TraceError("Error while collecting telemetry information: {0}", ex);
```